homeassistant/components/synology_dsm/update.py

```diff
@@ -0,0 +1,79 @@
+"""Support for Synology DSM update platform."""
+from __future__ import annotations
+
+from typing import Final
+
+from synology_dsm.api.core.upgrade import SynoCoreUpgrade
+from yarl import URL
+
+from homeassistant.components.update import UpdateEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SynoApi, SynologyDSMBaseEntity
+from .const import (
+    COORDINATOR_CENTRAL,
+    DOMAIN,
+    SYNO_API,
+    SynologyDSMUpdateEntityEntityDescription,
+)
+
+UPDATE_ENTITIES: Final = [
+    SynologyDSMUpdateEntityEntityDescription(
+        api_key=SynoCoreUpgrade.API_KEY,
+        key="update",
+        name="DSM Update",
+        entity_category=EntityCategory.CONFIG,
```

Comment:
```suggestion
```
This does not have any control (install) so should be diagnostics (which is the default when not providing a category)?

Reply:
default for `UpdateEntityDescription` is `EntityCategory.CONFIG` - see:
https://github.com/home-assistant/core/blob/945028d43d7bad4c3702037d57d8c9005f00edbc/homeassistant/components/update/__init__.py#L144-L149

Reply:
Yeah, was thinking about when not using description 👍 
I still think it should still be diagnostic

Reply:
TBH ... prior the `update`  entity exists, there was a binary sensor which shows if there is an update (_this was mostly a diagnostic entity_) and there was a button to perform the update (_this was most likely a config entity_) ... no both are combined into one entity ... so throw a coin to decide if it is diag or config 😄

Reply:
If the entity support install it should be config, if not it should be diagnostic

Reply:
ok, you convinced me 🙂